src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3973,10 +3973,9 @@ qreal ViewProviderSketch::getDevicePixelRatio() const
 {
     if (auto activeView = qobject_cast<Gui::View3DInventor*>(this->getActiveView())) {
         auto glWidget = activeView->getViewer()->getGLWidget();
-        return glWidget->devicePixelRatio();
+        return glWidget->physicalDpiX() / glWidget->logicalDpiX();
     }
-
-    return QApplication::primaryScreen()->devicePixelRatio();
+    return QApplication::primaryScreen()->physicalDotsPerInch() / QApplication::primaryScreen()->logicalDotsPerInch();
```

Comment:
Should this not be `glWidget->screen()` instead of `QApplication::primaryScreen()` to handle the details from the screen the widget is actually being drawn on?